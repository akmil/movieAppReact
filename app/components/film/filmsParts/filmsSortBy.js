import React from 'react';

//Films by the same genre
export const FilmsSortBy = ({genre}) => (
    <React.Fragment>
        Films by {genre} genre.
    </React.Fragment>
);

