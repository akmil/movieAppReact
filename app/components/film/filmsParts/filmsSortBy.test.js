import React from 'react';
import { FilmsSortBy, getVisibleMovie } from './filmsSortBy';
import renderer from 'react-test-renderer';

describe('<films FilmsSortBy/>', () => {
  it('should render <FilmsSortBy /> components', () => {
    // const wrapper = renderer.create(<FilmsSortBy/>);
    // expect(wrapper).toMatchSnapshot();
  });
  it('getVisibleMovie test', () => {
    const movies = [{
      name: 'test',
      genre: true,
    }, {
      name: 'test2',
      genre: false,
    }];
    const wrapperRes = getVisibleMovie(movies, 'SHOW_ALL');
    expect(wrapperRes).toEqual(movies);
    const wrapperRes2 = getVisibleMovie(movies, 'SHOW_ACTIVE');
    expect(wrapperRes2).toEqual(movies.filter(t => !t.genre));
  });
  it('getVisibleMovie test SHOW_DRAMA/SHOW_COMEDY', () => {
    const movies = [{
      name: 'test',
      genre: 'Drama2',
    }, {
      name: 'test2',
      genre: 'Comedy',
    }];
    const wrapperRes3 = getVisibleMovie(movies, 'SHOW_COMEDY');
    const wrapperRes4 = getVisibleMovie(movies, 'SHOW_DRAMA');

    expect(wrapperRes3).toEqual([{
      name: 'test2',
      genre: 'Comedy',
    }]);
    expect(wrapperRes4).toEqual([{
      name: 'test',
      genre: 'Drama2',
    }]);
  });
});
