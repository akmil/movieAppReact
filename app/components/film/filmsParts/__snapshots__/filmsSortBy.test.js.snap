// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<films FilmsSortBy/> should render <FilmsSortBy /> components 1`] = `
Array [
  "Films by ",
  " genre.",
]
`;
